aspnetcore/web-api/define-controller.md

```diff
@@ -0,0 +1,113 @@
+
+::: moniker range="<= aspnetcore-2.0"
+* [Derive class from Controller](#derive-class-from-controller)
+* [Derive class from ControllerBase](#derive-class-from-controllerbase)
+::: moniker-end
+::: moniker range=">= aspnetcore-2.1"
+* [Derive class from Controller](#derive-class-from-controller)
+* [Derive class from ControllerBase](#derive-class-from-controllerbase)
+* [Annotate class with ApiControllerAttribute](#annotate-class-with-apicontrollerattribute)
+::: moniker-end
+
+This document explains when it's most appropriate to use each option.
+
+## Derive class from Controller
+
+Inherit from the [Controller](/dotnet/api/microsoft.aspnetcore.mvc.controller) class when your controller needs to support presentation layer concerns in addition to Web API actions. Examples of presentation layer concerns include returning MVC views or [invoking view components](xref:mvc/views/view-components#invoking-a-view-component-directly-from-a-controller).
+
+[!code-csharp[](../web-api/define-controller/samples/WebApiSample.Api.Pre21/Controllers/OrdersController.cs?name=snippet_OrdersController&highlight=1)]
+
+In the preceding controller, the `Index` action returns a [ViewResult](/dotnet/api/microsoft.aspnetcore.mvc.viewresult) representing the associated MVC view at *Views/Orders/Index.cshtml*. The `GetById` and `CreateAsync` actions respond to HTTP GET and POST requests, respectively.
+
+## Derive class from ControllerBase
+
+Inherit from the [ControllerBase](/dotnet/api/microsoft.aspnetcore.mvc.controllerbase) class when your controller doesn't need to return MVC views or [invoke view components]((xref:mvc/views/view-components#invoking-a-view-component-directly-from-a-controller)). For example, the following controller only supports Web API actions:
+
+[!code-csharp[](../web-api/define-controller/samples/WebApiSample.Api.Pre21/Controllers/PetsController.cs?name=snippet_PetsController&highlight=3)]
+
+A benefit of deriving from `ControllerBase` instead of `Controller` is that IntelliSense displays only Web API-specific members.
+
+::: moniker range=">= aspnetcore-2.1"
+## Annotate class with ApiControllerAttribute
+
+ASP.NET Core 2.1 introduces the `[ApiController]` attribute to denote a Web API controller class. For example:
+
+[!code-csharp[](../web-api/define-controller/samples/WebApiSample.Api/Controllers/ProductsController.cs?name=snippet_ControllerSignature&highlight=2)]
+
+This attribute is commonly coupled with either `ControllerBase` or `Controller` to gain access to useful methods and properties. `ControllerBase` provides access to methods such as [CreatedAtAction](/dotnet/api/microsoft.aspnetcore.mvc.controllerbase.createdataction) and [File](/dotnet/api/microsoft.aspnetcore.mvc.controllerbase.file). `Controller` provides access to methods such as [Json](/dotnet/api/microsoft.aspnetcore.mvc.controller.json) and [View](/dotnet/api/microsoft.aspnetcore.mvc.controller.view).
```

Comment:
You can also create your own base class and slap `[ApiController]` on it. There's nothing magic about `Controller` or `ControllerBase`.

Reply:
> You can also create your own base class and slap `[ApiController]` on it. There's nothing magic about `Controller` or `ControllerBase`.

Can you suggest why you'd want to do that?

Reply:
Sure, lots of folks like to do one of two things:
1. Give us lots of specific feedback about specific action-result-creating methods they want to see or change
2. Create an `ApiController` **base class** because they like that idea

Users can solve both of these problems for themselves. They can create their own base class that either inherits from `ControllerBase`, slap `[ApiController]` on it and they are good to go.